profile-widget/ruleritem.cpp

```diff
@@ -152,11 +152,11 @@ RulerNodeItem2 *RulerItem2::destNode() const
 	return dest;
 }
 
-void RulerItem2::setPlotInfo(plot_info info)
+void RulerItem2::setPlotInfo(plot_info *info)
```

Comment:
For future reference: The idiomatic way would have been const reference + member initialization list:
```
void RulerItem2::setPlotInfo(const plot_info &info) : pInfo(info)
{
...
}
```

Avoids changes to the callers, communicates intent, constructs objects directly (no empty initialization + copy).

Reply:
Thanks a lot. Indeed, I have to learn idiomatic C++. I had tried simply adding a '&' but for some reason the compiler did not like that, so I went for the C way...

Reply:
No worries - it works, so let's keep it this way.

Reply:
I love learning new things. If this is the idiomatic way, then maybe we should switch to it anyway. Especially as many people here, OK, especially since AT LEAST I constantly look at "how did we do this elsewhere" when I don't know the idiomatic way to do things.
So if you have a moment, a cleanup PR for this one would be quite welcome, @bstoeger

Reply:
Ouch - my comment above was of course stupid, because `RulerItem2::setPlotInfo` is not a constructor. Apparently I was still sleeping.
Let's turn it into a const reference anyway for consistency with `RulerNodeItem2::setPlotInfo`.